src/viam/sdk/common/client_helper.hpp

```diff
@@ -63,21 +77,67 @@ class ClientHelper {
         client_helper_details::errorHandlerReturnedUnexpectedly(result);
     }
 
+    // A version of invoke for gRPC calls returning `(stream ResponseType)`.
+    // ResponseHandlerCallable will be called for every response in the reader, and should return
+    // false to indicate it is no longer interested in the stream.
+    template <typename ResponseHandlerCallable,
+              typename ErrorHandlerCallable = decltype(default_ehc_)>
+    auto invoke_stream(ResponseHandlerCallable rhc, ErrorHandlerCallable&& ehc = default_ehc_) {
+        *request_.mutable_name() = client_->name();
+        ClientContext ctx;
+
+        auto reader = (stub_->*pfn_)(ctx, request_);
+
+        bool cancelled_by_handler = false;
+
+        while (reader->Read(&response_)) {
+            if (!rhc(response_)) {
+                cancelled_by_handler = true;
+                static_cast<::grpc::ClientContext*>(ctx)->TryCancel();
```

Comment:
Why do you need the static_cast here?

Reply:
It's so I can call `grpc::ClientContext::TryCancel`, because `ClientContext ctx` above is `viam::sdk::ClientContext` which wraps the grpc context as a private member, and whose only public interface is a user-defined implicit conversion to `grpc::ClientContext*`, presumably so the stub method can be called without any explicit casts or operators

Reply:
Got it. Thanks for explaining. I knew that once but forgot. I may have even suggested it be that way...